src/sd/mmc.rs

```diff
@@ -0,0 +1,191 @@
+    ) -> Self {
+        Self {
+            slot: Slot::_0,
+            configuration: sdmmc_slot_config_t {
+                width: Self::get_slot_width(_d1, _d2, _d3, _d4, _d5, _d6, _d7),
+                flags: 0,
+                __bindgen_anon_1: sdmmc_slot_config_t__bindgen_ty_1 {
+                    cd: SDMMC_SLOT_NO_CD,
+                },
+                __bindgen_anon_2: sdmmc_slot_config_t__bindgen_ty_2 {
+                    wp: SDMMC_SLOT_NO_WP,
+                },
+            },
+            _p: PhantomData,
+        }
+    }
+
+    #[cfg(not(esp_idf_soc_sdmmc_use_gpio_matrix))]
+    pub fn new_slot_1(
+        _cmd: impl Peripheral<P = gpio::Gpio15> + 'd,
+        _clk: impl Peripheral<P = gpio::Gpio14> + 'd,
+        _d0: impl Peripheral<P = gpio::Gpio2> + 'd,
+        _d1: Option<impl Peripheral<P = gpio::Gpio4> + 'd>,
+        _d2: Option<impl Peripheral<P = gpio::Gpio12> + 'd>,
+        _d3: Option<impl Peripheral<P = gpio::Gpio13> + 'd>,
+    ) -> Self {
+        Self {
+            slot: Slot::_1,
+            configuration: sdmmc_slot_config_t {
+                width: Self::get_slot_width(
+                    _d1,
+                    _d2,
+                    _d3,
+                    Option::<()>::None,
+                    Option::<()>::None,
+                    Option::<()>::None,
+                    Option::<()>::None,
+                ),
+                flags: 0,
+                __bindgen_anon_1: sdmmc_slot_config_t__bindgen_ty_1 {
+                    gpio_cd: SDMMC_SLOT_NO_CD,
+                },
+                __bindgen_anon_2: sdmmc_slot_config_t__bindgen_ty_2 {
+                    gpio_wp: SDMMC_SLOT_NO_WP,
+                },
+            },
+            _p: PhantomData,
+        }
+    }
+
+    fn get_slot_width<A, B, C, D, E, F, G>(
+        _d1: Option<A>,
+        _d2: Option<B>,
+        _d3: Option<C>,
+        _d4: Option<D>,
+        _d5: Option<E>,
+        _d6: Option<F>,
+        _d7: Option<G>,
+    ) -> u8 {
+        if _d3.is_some() {
+            if !(_d2.is_some() && _d1.is_some()) {
+                panic!("D2 and D1 must be provided if D3 is provided");
+            }
+
+            if _d7.is_some() {
+                if !(_d6.is_some() && _d5.is_some() && _d4.is_some()) {
+                    panic!("D6, D5, and D4 must be provided if D7 is provided");
+                }
+
+                8
+            } else {
+                4
+            }
+        } else {
+            1
+        }
+    }
+
+    pub fn set_card_detect_pin(mut self, pin: impl Peripheral<P = impl InputPin> + 'd) -> Self {
+        self.configuration.__bindgen_anon_1.gpio_cd = pin.into_ref().deref().pin();
+        self
+    }
+
+    pub fn set_write_protect_pin(mut self, pin: impl Peripheral<P = impl InputPin> + 'd) -> Self {
+        self.configuration.__bindgen_anon_2.gpio_wp = pin.into_ref().deref().pin();
+        self
+    }
+
+    pub fn get_inner(&self) -> &sdmmc_slot_config_t {
```

Comment:
Why pub?

Reply:
Since the FAT (esp_vfs_fat_sdspi_mount) system requires MMC device configuration to mount the card.

Reply:
`pub(crate)` then please.